Documentation/dev/deploy-metering.md

```diff
@@ -0,0 +1,165 @@
+
+- `$METERING_NAMESPACE`/`--namespace`: the namespace to deploy metering resources on.
+- `$METERING_CR_FILE`/`--metering-cr`: the relative (or absolute) path to a customized `MeteringConfig` custom resource.
+- `$DEPLOY_MANIFESTS_DIR`/`--deploy-manifests-dir`: the relative (or absolute) path to the `manifests/deploy/` directory. Set this when working outside of the operator-metering cloned repository, or when the `bin/deploy-metering` binary has been moved to the user's `/usr/local/bin` directory.
+- `$DEPLOY_PLATFORM`/`--deploy-platform`: the platform in which metering is deploy on. The supported values include "upstream", "openshift", and "ocp-testing".
+
+#### Install-Only Configuration Options
+
+- `$SKIP_METERING_OPERATOR_DEPLOYMENT`/`--skip-metering-operator-deployment`: create only the `$METERING_NAMESPACE` namespace, the metering CRDs, and the `MeteringConfig` CR.
+- `$METERING_OPERATOR_IMAGE_REPO`/`--repo`: override the image repository used in the metering-ansible-operator container images.
+- `$METERING_OPERATOR_IMAGE_TAG`/`--tag`: override the image tag used in the metering-ansible-operator container images.
+
+**Note**: You need to set both the image repository and tag to override any of the container images.
+
+#### Uninstall-Only Configuration Options
+
+- `$METERING_DELETE_NAMESPACE`/`--delete-namespace`: defaults to false
+- `$METERING_DELETE_CRDS`/`--delete-crd`: defaults to false
+- `$METERING_DELETE_CRB`/`--delete-crb`: defaults to false
+- `$METERING_DELETE_PVCS`/`--delete-pvc`: defaults to true
+- `$METERING_DELETE_ALL`/`--delete-all`: defaults to false. This sets all the internal states that rely on the value of the ENV vars listed above, to true.
+
+### Example of precedence
+
+```bash
+$ export METERING_NAMESPACE="metering"
+$ ./bin/deploy-metering install --namespace test
+INFO[09-17-2019 14:54:42] Setting the log level to info                 app=deploy
+INFO[09-17-2019 14:54:42] Metering Deploy Namespace: test               app=deploy
+...
+```
+
+### Install Usage Examples
+
+**Note**: You can either use the flags, or the environment variables when running any of the following examples. Also, specify a customized `MeteringConfig`  custom resource before running any of these examples.
+
+#### Vanilla install on Openshift/OCP
+
+```bash
+export METERING_NAMESPACE="metering"
+./bin/deploy-metering install
+```
+
+#### Vanilla install on upstream
+
+```bash
+export METERING_NAMESPACE="metering"
+export DEPLOY_PLATFORM="upstream"
+./bin/deploy-metering install
+```
+
+#### Skip installing the metering operator
+
+```bash
+export METERING_NAMESPACE="metering"
+export SKIP_METERING_OPERATOR_DEPLOYMENT="true"
+./bin/deploy-metering install
+```
+
+#### Override the default images install
+
+```bash
+export METERING_NAMESPACE="metering"
+export METERING_OPERATOR_IMAGE_REPO="quay.io/tflannag/origin-metering-ansible-operator"
+export METERING_OPERATOR_IMAGE_TAG="latest"
+./bin/deploy-metering install
+```
```

Comment:
Let's make this more generic throughout.
export METERING_OPERATOR_IMAGE_REPO="quay.io/REPLACE-WITH-USERNAME/origin-metering-ansible-operator"
export METERING_OPERATOR_IMAGE_TAG="REPLACE-WITH-IMAGE-TAG"
or something less literal